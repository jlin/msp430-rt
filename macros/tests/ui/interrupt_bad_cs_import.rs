#![no_main]

use msp430::interrupt::CriticalSection as CritSec;
use msp430_rt_macros::{entry, interrupt};

// If we use "msp430::interrupt::CriticalSection", we get unused import warning for CritSec,
// which is correct but misleading b/c of interrupt sig requirements.
fn arg(_cs: msp430::interrupt::CriticalSection) {
    /* initialize hardware */
}

#[entry]
fn main() -> ! {
    unimplemented!()
}

#[interrupt]
fn TIM2(_cs: CritSec) {
    unimplemented!()
}
