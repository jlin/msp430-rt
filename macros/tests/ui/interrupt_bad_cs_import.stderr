error: `#[interrupt]` handlers must have signature `[unsafe] fn([<name>: CriticalSection]) [-> !]`
  --> tests/ui/interrupt_bad_cs_import.rs:18:1
   |
18 | fn TIM2(_cs: CritSec) {
   | ^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `msp430::interrupt::CriticalSection as CritSec`
 --> tests/ui/interrupt_bad_cs_import.rs:3:5
  |
3 | use msp430::interrupt::CriticalSection as CritSec;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
